docs/developer-tools/Guard.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Guard
+author: Sergio0694
+description: Helper methods to verify conditions when running code
+keywords: windows 10, uwp, windows community toolkit, uwp community toolkit, uwp toolkit, debug, net core, net standard
+dev_langs:
+  - csharp
+---
+
+# Guard
+
+The [Guard](https://docs.microsoft.com/dotnet/api/microsoft.toolkit.diagnostics.guard) can be used to validate method arguments in a streamlined manner, which is also faster, less verbose, more expressive and less error prone than manually writing checks and throwing exceptions.
+
+## How it works
+
+These `Guard` APIs are built with three core principles in mind:
+
+- Being **fast**. To achieve that, all the APIs have been designed to produce as little code as possible in the caller, and each single Guard API will (almost always) be inlined. Furthermore, specialized methods are generated with T4 templates to achieve the most efficient assembly code possible when working with common types (eg. primitive numeric types).
+- Being **helpful**. Each `Guard` API produces a detailed exception message that clearly specifies what went wrong, along with additional info (eg. current variable values), when applicable.
+- Being **intuitive**. To achieve this, all the `Guard` APIs have expressive and self-explanatory names that make it immediately clear what each API is supposed to do. This shifts the burden of writing checks away from the developers, letting them express conditions using natural language.
+
+## Syntax
+
+Here is a sample method, with some checks being done with explicit checks and manual throw statements:
+
+```csharp
+public static void SampleMethod(int[] array, int index, Span<int> span, string text)
+{
+    if (array is null)
+    {
+        throw new ArgumentNullException(nameof(array), "The array must not be null");
+    }
+
+    if (array.Length >= 10)
+    {
+        throw new ArgumentException($"The array must have less than 10 items, had a size of {array.Length}", nameof(array));
+    }
+
+    if (index < 0 || index >= array.Length)
+    {
+        throw new ArgumentOutOfRangeException(nameof(index), $"The index must be in the [0, {array.Length}) range, was {index}");
+    }
+
+    if (span.Length < array.Length)
+    {
+        throw new ArgumentException($"The target span is shorter than the input array, had a length of {span.Length}", nameof(span));
+    }
+
+    if (string.IsNullOrEmpty(text))
+    {
+        throw new ArgumentException("The input text can't be null or empty", nameof(text));
+    }
+}
+```
+
+And here is the same method, but using the new `Guard.APIs` to validate the input arguments:
+
+```csharp
+public static void SampleMethod(int[] array, int index, Span<int> span, string text)
+{
+    Guard.IsNotNull(array, nameof(array));
+    Guard.HasSizeAtLeast(array, 10, nameof(array));
+    Guard.IsInRangeFor(index, array, nameof(index));
+    Guard.HasSizeLessThanOrEqualTo(array, span, nameof(span));
+    Guard.IsNotNullOrEmpty(text, nameof(text));
```

Comment:
Make sure at least that all these examples used are in the table below.

Reply:
Fixed in https://github.com/MicrosoftDocs/WindowsCommunityToolkitDocs/pull/302/commits/3e9f73f7032c66bec5f1efea683d366b57088046.